core/services/ocr2/plugins/promwrapper/plugin.go

```diff
@@ -0,0 +1,141 @@
+package promwrapper
+
+import (
+	"context"
+	"math/big"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/smartcontractkit/libocr/offchainreporting2/types"
+)
+
+var _ types.ReportingPlugin = &promPlugin{}
+
+var (
+	promQuery = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_query_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Query() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promObservation = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_observation_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Observation() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Report() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldAcceptFinalizedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_accept_finalized_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldAcceptFinalizedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promShouldTransmitAcceptedReport = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_should_transmit_accepted_report_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's ShouldTransmitAcceptedReport() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
+	)
+	promClose = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "ocr2_reporting_plugin_close_duration",
+			Help:    "The amount of time elapsed during the OCR2 plugin's Close() method",
+			Buckets: []float64{}, // TODO: figure out buckets
+		},
+		[]string{"evmChainID", "pluginName"},
```

Comment:
EVM only? Or would generalizing a bit make sense?
```suggestion
		[]string{"chainType", "chainID", "pluginName"},
```

Reply:
yes generalizing would make sense here